File: MyTodos.Api/Models/Todo.cs
namespace MyTodos.Api.Models;

public class Todo
{
    public Guid Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public DateOnly? DueDate { get; set; }
    public bool IsComplete { get; set; }
    public DateTime CreatedAt { get; set; }
}

File: mytodos-frontend/src/types/todo.ts
export interface Todo {
  id: string
  title: string
  dueDate: string | null
  isComplete: boolean
  createdAt: string
}

export interface CreateTodoRequest {
  title: string
  dueDate?: string | null
}

export interface UpdateTodoRequest {
  title: string
  dueDate?: string | null
  isComplete: boolean
}
